### PR TITLE
Add detailed error logging in case of failed blocks download

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

Prior to this PR, a failed block download during `processStageBlocksDownload` would not get logged. Instead, it would retry to download the block following the configured `cs.config.CatchupBlockDownloadRetryAttempts` limits.

This PR adds an info logging in case of such a failure. The intent of this PR is to provide the node runner with the information about the download error so that she would be able to take concrete actions around that specific failure.

## Test Plan

Tested manually.

### CODE REVIEW

File: catchup/catchpointService.go [hunk 1]
```diff
@@ -466,6 +466,7 @@ func (cs *CatchpointCatchupService) processStageBlocksDownload() (err error) {
 				}
 				if attemptsCount <= uint64(cs.config.CatchupBlockDownloadRetryAttempts) {
 					// try again.
+					cs.log.Infof("Failed to download block %d. %v", topBlock.Round()-basics.Round(blocksFetched), err)
```
Comment (pbennett) [inline catchup/catchpointService.go:469]: Perhaps show attempt x of y in message    
ie:
Failed to download block %d (attempt %d of %d). %v

File: catchup/httpFetcher.go [hunk 2]
```diff
@@ -100,8 +100,13 @@ func (hf *HTTPFetcher) GetBlockBytes(ctx context.Context, r basics.Round) (data
 		return nil, errNoBlockForRound
 	default:
 		hf.log.Warn("http block fetcher response status code : ", response.StatusCode)
-		response.Body.Close()
```
Comment (algonautshant) [inline catchup/httpFetcher.go:103]: why response Body is no longer closed in this case?

Comment (algonautshant) [reply catchup/httpFetcher.go:103]: because it is closed in rpcs.ResponseBytes :-)

**Review by algonautshant — APPROVED** [review]
Looks good.

File: catchup/catchpointService.go (hunk 1 above)
Comment (tsachiherman) [reply catchup/catchpointService.go:469]: I don't know if that's a super useful piece of information here, but I don't mind adding it either.

**Review by algonautshant — APPROVED** [review]

**Pull request merged**